tests/by-util/test_expr.rs

```diff
@@ -518,6 +518,31 @@ fn test_substr_precedence() {
         .args(&["substr", "ab cd", "2", "1", "!=", " "])
         .succeeds()
         .stdout_only("1\n");
+
+    new_ucmd!()
+        .args(&["length", "abcd", "!=", "4"])
+        .fails_with_code(1)
+        .stdout_only("0\n");
+
+    new_ucmd!()
+        .args(&["length", "abcd", "=", "4"])
+        .succeeds()
+        .stdout_only("1\n");
+
+    new_ucmd!()
+        .args(&["index", "abcd", "c", "!=", "2"])
+        .succeeds()
+        .stdout_only("1\n");
+
+    new_ucmd!()
+        .args(&["index", "abcd", "c", "=", "2"])
+        .fails_with_code(1)
+        .stdout_only("0\n");
```

Comment:
What's the reason you deviate from the pattern used for `substr` and `length`, where the commands with `=` are expected to succeed and the commands with `!=` are expected to fail?